Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.Revolver.cs

```diff
@@ -26,17 +28,17 @@ protected virtual void InitializeRevolver()
         SubscribeLocalEvent<RevolverAmmoProviderComponent, GetVerbsEvent<AlternativeVerb>>(OnRevolverVerbs);
         SubscribeLocalEvent<RevolverAmmoProviderComponent, InteractUsingEvent>(OnRevolverInteractUsing);
         SubscribeLocalEvent<RevolverAmmoProviderComponent, GetAmmoCountEvent>(OnRevolverGetAmmoCount);
-        SubscribeLocalEvent<RevolverAmmoProviderComponent, UseInHandEvent>(OnRevolverUse);
+        SubscribeLocalEvent<RevolverAmmoProviderComponent, UniqueActionEvent>(OnRevolverUniqueAction);
     }
 
-    private void OnRevolverUse(EntityUid uid, RevolverAmmoProviderComponent component, UseInHandEvent args)
+    private void OnRevolverUniqueAction(Entity<RevolverAmmoProviderComponent> gun, ref UniqueActionEvent args)
     {
-        if (!_useDelay.TryResetDelay(uid))
+        if (!_useDelay.TryResetDelay(gun)) // cannot wield after cycling. Can cycle after wielding tho? Can shoot regardless.
             return;
 
-        Cycle(component);
-        UpdateAmmoCount(uid, prediction: false);
-        Dirty(uid, component);
+        Cycle(gun.Comp);
+        UpdateAmmoCount(gun, prediction: false);
+        Dirty(gun, gun.Comp);
```

Comment:
This should be a field on the component that makes UseInHand happen or not, and the UniqueActionEvent a subscription in one of our systems for merge conflicts.